waku/waku_store/rpc_codec.nim

```diff
@@ -3,257 +3,210 @@ when (NimMajor, NimMinor) < (1, 4):
 else:
   {.push raises: [].}
 
-import std/options, nimcrypto/hash
-import ../common/[protobuf, paging], ../waku_core, ./common, ./rpc
+import std/options, stew/arrayops, nimcrypto/hash
+import ../common/[protobuf, paging], ../waku_core, ./common
 
+#TODO recalculate the max size
 const MaxRpcSize* = MaxPageSize * MaxWakuMessageSize + 64 * 1024
   # We add a 64kB safety buffer for protocol overhead
 
-## Pagination
+### Request ###
 
-proc encode*(index: PagingIndexRPC): ProtoBuffer =
-  ## Encode an Index object into a ProtoBuffer
-  ## returns the resultant ProtoBuffer
+proc encode*(req: StoreQueryRequest): ProtoBuffer =
```

Comment:
General Question:

We are manually encoding protobufs, seems error-prone and high maintenance. 
Can't we use something like https://github.com/PMunch/protobuf-nim to generate the nim code?

Reply:
We should but if I'm not mistaken we use proto2 and not 3?

Reply:
Maybe this can be taken up as an improvement as part of separate PR. 

cc @Ivansete-status

Reply:
Very interesting @chaitanyaprem ! I think that is something we could look at but I don't see super urgency for now as we don't change the protobuf implementation very often

wdyt @jm-clius ?